routes/transaction.go

```diff
@@ -72,6 +73,146 @@ func (fes *APIServer) GetTxn(ww http.ResponseWriter, req *http.Request) {
+		if err := signedInnerTxn.FromBytes(signedTxnBytes); err != nil {
+			_AddBadRequestError(ww, fmt.Sprintf(
+				"SubmitAtomicTransaction: Problem deserializing signed transaction %d from bytes: %v",
+				ii, err))
+			return
+		}
+
+		// Verify the signature is present.
+		if signedInnerTxn.Signature.Sign == nil {
+			_AddBadRequestError(ww, fmt.Sprintf(
+				"SubmitAtomicTransaction: Signed transaction %d hex missing signature", ii))
+			return
+		}
+
+		// Find the pre-signature DeSo transaction hash.
+		// NOTE: We do not use the lib.MsgDeSoTxn.Hash() function here as
+		// the transactions included in the atomic transaction do not yet
+		// have their signature fields set.
+		preSignatureInnerTxnBytes, err := signedInnerTxn.ToBytes(true)
+		if err != nil {
+			_AddBadRequestError(ww, fmt.Sprintf(
+				"SubmitAtomicTransaction: Problem serializing "+
+					"signed transaction %d without signature: %v", ii, err))
+			return
+		}
+		preSignatureInnerTxnHash := lib.Sha256DoubleHash(preSignatureInnerTxnBytes)
+		innerTxnPreSignatureHashToSignature[*preSignatureInnerTxnHash] = signedInnerTxn.Signature
+	}
+
+	// Based on the provided signatures, complete the atomic transaction.
+	for jj, innerTxn := range atomicTxn.TxnMeta.(*lib.AtomicTxnsWrapperMetadata).Txns {
+		// Skip signed inner transactions.
+		if innerTxn.Signature.Sign != nil {
+			continue
+		}
+
+		// Find the pre-signature DeSo transaction hash for this transaction.
+		preSignatureInnerTxnBytes, err := innerTxn.ToBytes(true)
+		if err != nil {
+			_AddBadRequestError(ww, fmt.Sprintf(
+				"SubmitAtomicTransaction: Problem serializing "+
+					"transaction %d of atomic transaction wrapper without signature: %v", jj, err))
+			return
+		}
+		preSignatureInnerTxnHash := lib.Sha256DoubleHash(preSignatureInnerTxnBytes)
+
+		// Check that we have the signature.
+		if _, exists := innerTxnPreSignatureHashToSignature[*preSignatureInnerTxnHash]; !exists {
+			_AddBadRequestError(ww, fmt.Sprintf(
+				"SubmitAtomicTransaction: Transaction %d in atomic transaction still missing signature", jj))
+			return
+		}
+
+		// Set the signature in the atomic transaction.
+		atomicTxn.TxnMeta.(*lib.AtomicTxnsWrapperMetadata).Txns[jj].Signature =
+			innerTxnPreSignatureHashToSignature[*preSignatureInnerTxnHash]
+	}
+
+	// Verify and broadcast the completed atomic transaction.
+	if err := fes.backendServer.VerifyAndBroadcastTransaction(atomicTxn); err != nil {
+		_AddBadRequestError(ww,
+			fmt.Sprintf("SubmitAtomicTransaction: Problem broadcasting transaction: %v", err))
+		return
+	}
+
+	res := &SubmitAtomicTransactionResponse{
+		Transaction:              atomicTxn,
+		TxnHashHex:               atomicTxn.Hash().String(),
+		TransactionIDBase58Check: lib.PkToString(atomicTxn.Hash()[:], fes.Params),
+	}
+
+	if err := json.NewEncoder(ww).Encode(res); err != nil {
+		_AddBadRequestError(ww, fmt.Sprintf(
+			"SubmitAtomicTransaction: Problem encoding response as JSON: %v", err))
+		return
+	}
+}
```

Comment:
Can we move this to a separate PR such that we can accept this change independently of the subsidization changes?

Reply:
Yes! I'm going to leave this in until @diamondhands0 has a chance to read through everything. 

Something we might consider is leaving this backend PR as a draft until we decide we need it. It shouldn't be too difficult to merge this PR in later if we wanted to. It's mostly just adding new endpoints and helper functions.